test/interval_tests/construction.jl

```diff
@@ -35,6 +35,10 @@ using Base.Test
     @test Interval{BigFloat}(1) == Interval{BigFloat}(big(1.0), big(1.0))
     @test Interval{BigFloat}(pi) == Interval{BigFloat}(big(pi), big(pi))
 
+    # a < Inf and b > -Inf
+    @test Interval(Inf, Inf) == Interval(prevfloat(Inf), Inf)
```

Comment:
I agree it should probably yield an error or an empty interval; however, due to `rationalize` in the convert method which is used, we get Interval(Inf,Inf) when it shouldn't. So for the time being I think we should focus in getting around rationalize.